recipes/ogre/1.x/conanfile.py

```diff
@@ -0,0 +1,320 @@
+        cmake.definitions["OGRE_BUILD_PLUGIN_BSP"] = self.options.ogre_build_plugin_bsp
+        cmake.definitions["OGRE_BUILD_PLUGIN_OCTREE"] = self.options.ogre_build_plugin_octree
+        cmake.definitions["OGRE_BUILD_PLUGIN_PCZ"] = self.options.ogre_build_plugin_pcz
+        cmake.definitions["OGRE_BUILD_PLUGIN_PFX"] = self.options.ogre_build_plugin_pfx
+        cmake.definitions["OGRE_BUILD_RENDERSYSTEM_D3D11"] = self.options.ogre_build_rendersystem_d3d11
+        cmake.definitions["OGRE_BUILD_RENDERSYSTEM_GL"] = self.options.ogre_build_rendersystem_gl
+        cmake.definitions["OGRE_BUILD_RENDERSYSTEM_GL3PLUS"] = self.options.ogre_build_rendersystem_gl3plus
+        cmake.definitions["OGRE_BUILD_SAMPLES"] = self.options.ogre_build_samples
+        cmake.definitions["OGRE_BUILD_TOOLS"] = self.options.ogre_build_tools
+        cmake.definitions["OGRE_CONFIG_ENABLE_QUAD_BUFFER_STEREO"] = self.options.ogre_config_enable_quad_buffer_stereo
+        cmake.definitions["OGRE_CONFIG_FILESYSTEM_UNICODE"] = self.options.ogre_config_filesystem_unicode
+        cmake.definitions["OGRE_CONFIG_THREADS"] = self.options.ogre_config_threads
+        cmake.definitions["OGRE_CONFIG_THREAD_PROVIDER"] = self.options.ogre_config_thread_provider
+        cmake.definitions["OGRE_CONFIG_ENABLE_FREEIMAGE"] = self.options.ogre_config_enable_freeimage
+        cmake.definitions["OGRE_INSTALL_PDB"] = self.options.ogre_install_pdb
+        cmake.definitions["OGRE_INSTALL_SAMPLES"] = self.options.ogre_install_samples
+        cmake.definitions["OGRE_INSTALL_TOOLS"] = self.options.ogre_install_tools
+        cmake.definitions["OGRE_RESOURCEMANAGER_STRICT"] = self.options.ogre_resourcemanager_strict
+        if self.settings.os == "Windows":
+            cmake.definitions["OGRE_INSTALL_VSPROPS"] = self.options.ogre_install_vsprops
+        cmake.configure()
+        return cmake
+
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def build(self):
+        # the pkgs below are not available as conan recipes yet
+        # TODO: delte line 200-208 once the conan recipes are available
+        ogre_pkg_modules = ["AMDQBS", "Cg", "HLSL2GLSL", "GLSLOptimizer", "OpenGLES", "OpenGLES2", "OpenGLES3", "SDL2", "Softimage", "Wix"]
+        ogre_pkg_module_path = os.path.join(self.build_folder, self._source_subfolder, "CMake", "Packages")
+        for pkg_module in ogre_pkg_modules:
+            pkg_path = os.path.join(ogre_pkg_module_path, f"Find{pkg_module}.cmake")
+            if os.path.isfile(pkg_path):
+                shutil.copy(pkg_path, self.build_folder)
+            else:
+                raise RuntimeError(f"The file Find{pkg_module}.cmake is not present in f{ogre_pkg_module_path}!")
+
+
```

Comment:
```suggestion
    def _patch_sources(self):
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)
        # the pkgs below are not available as conan recipes yet
        # TODO: delte line 200-208 once the conan recipes are available
        ogre_pkg_modules = ["AMDQBS", "Cg", "HLSL2GLSL", "GLSLOptimizer", "OpenGLES", "OpenGLES2", "OpenGLES3", "SDL2", "Softimage", "Wix"]
        ogre_pkg_module_path = os.path.join(self.build_folder, self._source_subfolder, "CMake", "Packages")
        for pkg_module in ogre_pkg_modules:
            pkg_path = os.path.join(ogre_pkg_module_path, f"Find{pkg_module}.cmake")
            if os.path.isfile(pkg_path):
                shutil.copy(pkg_path, self.build_folder)
            else:
                raise ConanException(f"The file Find{pkg_module}.cmake is not present in f{ogre_pkg_module_path}!")

    def build(self):
        self._patch_sources()
```

The method `source` should not apply/change anything. Otherwise, you will need to download again the entire source code for any minimal change.

Don't use generic Exception, prefer ConanException, which is treated by Conan client.